src/main/java/com/aws/greengrass/config/Topics.java

```diff
@@ -123,7 +123,8 @@ private Topic createLeafChild(CaseInsensitiveString name, long timestamp) {
         if (n instanceof Topic) {
             return (Topic) n;
         } else {
-            throw new IllegalArgumentException(name + " in " + this + " is already a container, cannot become a leaf");
+            throw new IllegalArgumentException(name + " in "
+                    + getFullName() + " is already a container, cannot become a leaf");
```

Comment:
should this be the parent node's key path?

Reply:
No, have a look at the error messages that I'm validating in the test.